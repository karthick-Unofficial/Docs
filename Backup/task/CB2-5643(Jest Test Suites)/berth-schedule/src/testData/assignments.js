const assignments = {
    "allAssignments": {},
    "dailyAssignments": {}
};

export { assignments };