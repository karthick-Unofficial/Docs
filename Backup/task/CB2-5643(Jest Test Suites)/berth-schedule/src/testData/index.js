export { appState } from "./appState";
export { session } from "./session";
export { i18n } from "./i18n";
export { map } from "./map";
export { berths } from "./berths";
export { formPanel } from "./formPanel";
export { assignments } from "./assignments";
export { clientConfig } from "./clientConfig";