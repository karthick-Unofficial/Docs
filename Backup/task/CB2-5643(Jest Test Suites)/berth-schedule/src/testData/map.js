const map = {
    "open": false,
    "vessels": {},
    "subscriptions": []
};
export { map };