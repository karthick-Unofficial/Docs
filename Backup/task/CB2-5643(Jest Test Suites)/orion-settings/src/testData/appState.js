const appState = {
    "errors": {
        "newUserErrorMessage": null,
        "passwordChangeError": null,
        "shareDialogError": null,
        "createOrgError": null
    },
    "dialog": {
        "openDialog": null,
        "dialogData": null
    },
    "dock": {
        "dockData": {
            "WavCam": {}
        }
    },
    "sharingTokens": {
        "enabled": true
    },
    "viewing": {
        "selectedEntity": {
            "type": "org",
            "id": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b"
        }
    }
};
export { appState };