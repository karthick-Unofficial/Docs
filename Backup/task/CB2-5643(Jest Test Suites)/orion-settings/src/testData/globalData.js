const globalData = {
    "orgs": {},
    "users": {
        "c1bc0cd8-9dc8-4dbb-99c1-a0300aa05700": {
            "admin": true,
            "appSettings": {
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                }
            },
            "authProviderId": "system",
            "contact": {
                "cellPhone": null,
                "officePhone": null
            },
            "createdDate": "2023-01-13T08:00:18.692Z",
            "ecoAdmin": false,
            "email": "testorg@gm.com",
            "failedLoginAttempts": 0,
            "firstUseAck": true,
            "id": "c1bc0cd8-9dc8-4dbb-99c1-a0300aa05700",
            "lastModifiedDate": "2023-01-13T08:05:16.333Z",
            "lockedUntil": null,
            "name": "Test Org",
            "orgId": "test_organization_1",
            "orgRole": {
                "roleId": null,
                "title": "Org Admin"
            },
            "passwordChangedDate": "2023-01-13T08:02:06.726Z",
            "passwordExpires": "2023-04-13T08:02:06.726Z",
            "profileImage": "03ba4d40-9319-11ed-b28d-4bd7f7aa52bf",
            "resetExpires": null,
            "resetToken": null,
            "role": "Org User",
            "roleId": "test_organization_1_org_user",
            "setPasswordToken": null,
            "username": "testorg@gm.com"
        },
        "d7d232cb-a192-4290-83b8-a70e744a27b0": {
            "admin": false,
            "appSettings": {
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                }
            },
            "authProviderId": "system",
            "contact": {
                "cellPhone": null,
                "officePhone": null
            },
            "createdDate": "2023-01-13T05:01:08.255Z",
            "ecoAdmin": false,
            "email": "s.karthick@rifluxyss.com",
            "failedLoginAttempts": 0,
            "firstUseAck": true,
            "id": "d7d232cb-a192-4290-83b8-a70e744a27b0",
            "lastModifiedDate": "2023-01-23T10:57:40.656Z",
            "lockedUntil": null,
            "name": "Karthick S",
            "orgId": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b",
            "orgRole": {
                "roleId": "org-user",
                "title": "Org User"
            },
            "passwordChangedDate": "2023-01-13T05:10:07.858Z",
            "passwordExpires": "2023-04-13T05:10:07.858Z",
            "profileImage": "c192ce50-9b0c-11ed-beb9-2dd4a7b49e21",
            "resetExpires": null,
            "resetToken": null,
            "role": "Org User",
            "roleId": "ares_security_corporation_org_user",
            "setPasswordToken": null,
            "username": "s.karthick@rifluxyss.com"
        },
        "9610acee-dd03-479c-ab02-5f8fcdb0409f": {
            "admin": true,
            "appSettings": {
                "locale": null,
                "spotlightProximity": {
                    "unit": "mi",
                    "value": 0.25
                },
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                },
                "unitsOfMeasurement": {
                    "coordinateSystem": "decimal-degrees",
                    "landUnitSystem": "imperial"
                }
            },
            "authProviderId": "system",
            "contact": {
                "address": "1501 Main Street, Suite 120 Columbia, SC 29201",
                "email": "orgadmin@aressecuritycorp.com",
                "phone": "803-339-0697"
            },
            "disabled": false,
            "ecoAdmin": false,
            "email": "orgadmin@aressecuritycorp.com",
            "id": "9610acee-dd03-479c-ab02-5f8fcdb0409f",
            "name": "Org Admin",
            "orgId": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b",
            "role": "org-admin",
            "roleId": "ares_security_corporation_org_admin",
            "username": "orgadmin@aressecuritycorp.com"
        },
        "ed3d2e80-9737-40cd-ad06-74dc8c6e2003": {
            "admin": true,
            "appSettings": {
                "locale": "en",
                "spotlightProximity": {
                    "unit": "mi",
                    "value": 0.25
                },
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                },
                "unitsOfMeasurement": {
                    "coordinateSystem": "decimal-degrees",
                    "landUnitSystem": "imperial"
                }
            },
            "authProviderId": "system",
            "contact": {
                "address": "1501 Main Street, Suite 120 Columbia, SC 29201",
                "cellPhone": null,
                "email": "ecoadmin@aressecuritycorp.com",
                "officePhone": null,
                "phone": "803-339-0697"
            },
            "disabled": false,
            "ecoAdmin": true,
            "email": "ecoadmin@aressecuritycorp.com",
            "failedLoginAttempts": 0,
            "firstUseAck": true,
            "id": "ed3d2e80-9737-40cd-ad06-74dc8c6e2003",
            "lastModifiedDate": "2023-01-23T06:53:35.286Z",
            "lockedUntil": null,
            "name": "Support Admin",
            "orgId": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b",
            "profileImage": "a83a6f20-9aea-11ed-9228-07434b76e195",
            "role": "eco-admin",
            "roleId": "ares_security_corporation_org_admin",
            "username": "ecoadmin@aressecuritycorp.com"
        }
    }
};
export { globalData };