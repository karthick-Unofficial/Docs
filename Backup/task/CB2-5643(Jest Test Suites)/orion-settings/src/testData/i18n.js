const i18n = {
	"locale": "en",
	"translations": {}
};
export { i18n };