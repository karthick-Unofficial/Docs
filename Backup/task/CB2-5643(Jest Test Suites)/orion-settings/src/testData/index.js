export { appState } from "./appState";
export { session } from "./session";
export { i18n } from "./i18n";
export { globalData } from "./globalData";