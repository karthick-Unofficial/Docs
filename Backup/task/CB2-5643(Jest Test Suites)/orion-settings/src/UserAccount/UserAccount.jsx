import React, { useEffect, Fragment, memo } from "react";
// router
import { useNavigate, useParams } from "react-router-dom";
import { routes as r } from "../routes.js";
// material-ui
import { Container, useMediaQuery } from "@mui/material";
// Components
import Profile from "./components/Profile";
import Settings from "./components/Settings";
import SideNavigation from "../shared/components/SideNavigation/SideNavigation";
import { UserApps, UserIntegrations } from "./components";
import { Translate } from "orion-components/i18n";
import EditPageTemplate from "../shared/components/EditPageTemplate";
import { useDispatch, useSelector } from "react-redux";
import * as actionCreators from "./userAccountActions";
import { getDir } from "orion-components/i18n/Config/selector";

const UserAccount = () => {
	const dispatch = useDispatch();
	const params = useParams();

	const {
		fetchUserProfile,
		updateGlobalUserAppSettings,
		setLocaleWithFallback
	} = actionCreators;

	// -- don't do anything until globalData is populated
	const users = useSelector(state => Object.keys(state.globalData.users).length === 0);

	// Read url parameters to  determine if this is the viewing user's profile or not, and
	// retrieve correct id from state
	const username = useSelector(state => !users ? (params && params.id || state.session.user.profile.id) : undefined);
	const user = useSelector(state =>
		!users ? (username ? state.globalData.users[username] :
			(state.appState.viewing.selectedEntity.type === "user" ? state.globalData.users[state.appState.viewing.selectedEntity.id] : null)) : undefined);
	const isMyUser = useSelector(state => !users ? (username === state.session.user.profile.id) : undefined);
	const isAdmin = useSelector(state => state.session.user.profile.admin);
	const isMyOrg = useSelector(state => user && state.session.user.profile.orgId === user.orgId);
	const org = useSelector(state => !users ? (user ? state.globalData.orgs[user.orgId] : null) : undefined);

	// Only users in the same organization can view each other
	const canViewThisUser = !users ? isMyUser || (isAdmin && isMyOrg) : undefined;
	const canEditThisUser = !users ? isMyUser || (isAdmin && isMyOrg && user && !user.admin) : undefined;
	// ------------------------------------------------------
	const globalState = useSelector(state => !users ? state.appState.global : undefined);
	const locale = useSelector(state => state.i18n.locale);
	// const WavCamOpen = useSelector(state => !users && state.appState.dock.dockData.WavCam);   Not used anymore
	const dir = useSelector(state => getDir(state));

	const navigate = useNavigate();

	useEffect(() => {
		if (username) {
			if (canViewThisUser === false || !user) {
				navigate(r.MY_ACCOUNT, { replace: true });
			} else if (username && fetchUserProfile) {
				dispatch(fetchUserProfile(username));
			}
		}
	}, [username, fetchUserProfile, canViewThisUser]);
	const minWidth1024Query = useMediaQuery("(min-width:1024px)");
	const minWidth720Query = useMediaQuery("(min-width:720px)");
	const minWidth600Query = useMediaQuery("(min-width:600px)");
	const mixedQuery = useMediaQuery("(max-width:1023px) and (min-width:720px), (min-width:1048px)");
	if (!user) return null;

	const styles = {
		accountContent: {
			...(dir == "rtl" ? { marginRight: minWidth1024Query ? 280 : 0 } : { marginLeft: minWidth1024Query ? 280 : 0 })
		}
	};

	const userAccountContainer = () => {
		return (
			<div>
				{/* Navigation */}
				{isMyUser && <SideNavigation location={1} />}
				{/* Main User Account Content */}
				<div style={styles.accountContent}>
					<div style={{
						marginLeft: "auto",
						marginRight: "auto",
						maxWidth: isMyUser ? minWidth1024Query ? 1120 : 840 : minWidth1024Query ? 1040 : 750,
						paddingLeft: minWidth1024Query ? 48 : minWidth720Query ? 24 : minWidth600Query ? 16 : 8,
						paddingRight: minWidth1024Query ? 48 : minWidth720Query ? 24 : minWidth600Query ? 16 : 8
					}}>
						<div style={{
							maxWidth: 840
						}}>
							{/* My Settings Header */}
							<header style={{ textAlign: "center" }}>
								<h2>
									{/* {isMyUser ? "My" : "User"} Account Settings */}
									<Translate value={isMyUser ? "sideBar.option1.variant1" : "sideBar.option1.variant2"} />
								</h2>
								<div className="b1-dark-gray">
									<Translate value="mainContent.accountSettings.titleText" />
								</div>
							</header>
							<Container component={"section"} style={{
								paddingLeft: 0,
								paddingRight: 0,
								paddingTop: mixedQuery ? 0 : minWidth600Query ? 8 : 16,
								paddingBottom: 20,
								...(mixedQuery ? {
									boxAlign: "stretch",
									alighnItems: "stretch",
									display: "flex",
									flexWrap: "wrap",
									marginLeft: -12,
									width: "calc(100% + 24px)"
								} : {})
							}}>
								{user && (
									<Fragment>
										{/* User Profile */}
										<Profile user={user} org={org} isMyUser={isMyUser} canEditThisUser={canEditThisUser} dir={dir} />
										{/* User Settings */}
										{isMyUser && <Settings globalState={globalState} updateGlobalUserAppSettings={updateGlobalUserAppSettings} setLocaleWithFallback={setLocaleWithFallback} dir={dir} lngLocale={locale} />}
										{/* User Apps */}
										{/* <UserApps isMyUser={isMyUser} /> */}
										{/* User Integrations */}
										{/* <UserIntegrations isMyUser={isMyUser} /> */}
									</Fragment>
								)}
							</Container>
						</div>
					</div>
				</div>
			</div >);
	};
	return (
		!isMyUser ? <EditPageTemplate
			title={<Translate value="mainContent.manageOrganization.orgSettings.manageUsers.title" />}
			dir={dir}
		>
			{userAccountContainer()}
		</EditPageTemplate> :
			<div>
				{userAccountContainer()}
			</div>
	);
};

export default memo(UserAccount);