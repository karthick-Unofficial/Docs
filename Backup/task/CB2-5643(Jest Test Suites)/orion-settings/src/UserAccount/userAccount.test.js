import React from "react";
import UserAccount from "./UserAccount";
import { shallow } from "enzyme";
import * as reactRedux from 'react-redux';
import { appState, i18n, session, globalData } from "../testData";

jest.mock("react-redux", () => ({
	useSelector: jest.fn(),
	useDispatch: jest.fn(),
}));

jest.mock('orion-components', () => ({
	getTranslation: () => { },
	getDir: () => { }
}));

jest.mock('react-router-dom', () => ({
	useNavigate: jest.fn(),
	useParams: jest.fn()
}));

describe("UserAccount", () => {
	beforeEach(() => {
		useSelectorMock.mockImplementation(selector => selector(mockStore));
	})
	afterEach(() => {
		useDispatchMock.mockClear();
		useSelectorMock.mockClear();
	})

	const useSelectorMock = reactRedux.useSelector;
	const useDispatchMock = reactRedux.useDispatch;

	const mockStore = {
		appState,
		session,
		i18n,
		globalData
	};

	it("renders", () => {
		const wrapper = shallow(<UserAccount />);
		expect(wrapper).toMatchSnapshot();
	});
});