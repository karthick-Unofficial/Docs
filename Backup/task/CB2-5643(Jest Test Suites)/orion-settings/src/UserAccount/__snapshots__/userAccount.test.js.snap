// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAccount renders 1`] = `ShallowWrapper {}`;
